File: labServiceUIJava/labServiceUIJava/src/main/java/com/example/labserviceuijava/repository/OrderDBRepository.java
package com.example.labserviceuijava.repository;

import com.example.labserviceuijava.model.Order;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.Properties;

import com.example.labserviceuijava.repository.exceptions.RepositoryException;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class OrderDBRepository implements OrderRepository{

    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public OrderDBRepository(Properties properties){
        logger.info("Initializing ConcertDBRepository with properties: {} ",properties);
        dbUtils=new JdbcUtils(properties);
    }

    @Override
    public Order findOne(Integer integer) {
        return null;
    }

    @Override
    public Iterable<Order> findAll() {
        return null;
    }

    @Override
    public Order save(Order entity) {
        logger.traceEntry("saving order task{}");
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preparedStatement=con.prepareStatement("insert into orders(concertID,buyerName,numberOfTickets) values (?,?,?)")){
            preparedStatement.setInt(1,entity.getConcertId());
            preparedStatement.setString(2,entity.getBuyerName());
            preparedStatement.setInt(3,entity.getNumberOfTickets());
            int result=preparedStatement.executeUpdate();
            logger.trace("saved {} objects",result);
        }catch (SQLException e){
            logger.error(e);
            System.err.println("ERROR DB" + e);
        }
        logger.traceExit(entity);
        return entity;
    }

    @Override
    public Order delete(Integer integer) {
        return null;
    }

    @Override
    public Order update(Order entity) {
        return null;
    }
}

File: labSocketsFestivalCSV2/ConsoleApplication1/persistence/OrderDBRepository.cs
using System;
using System.Collections.Generic;
using model;

namespace persistence
{
    public class OrderDBRepository:IOrderRepository
    {
        private IDictionary<string, string> props;

        public OrderDBRepository(IDictionary<string, string> props)
        {
            this.props = props;
        }
        
        public void Save(Order entity)
        {
            var con = DbUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into order(concertId,buyerName,numberOfTickets) values(@cId,@bName,@nr)";
                var paramCId = comm.CreateParameter();
                paramCId.ParameterName = "@cId";
                paramCId.Value = entity.ConcertId;
                comm.Parameters.Add(paramCId);

                var paramBName = comm.CreateParameter();
                paramBName.ParameterName = "@bName";
                paramBName.Value = entity.BuyerName;
                comm.Parameters.Add(paramBName);

                var paramNr = comm.CreateParameter();
                paramNr.ParameterName = "@nr";
                paramNr.Value = entity.NumberOfTickets;
                comm.Parameters.Add(paramNr);

                int rez = comm.ExecuteNonQuery();
                Console.WriteLine("MessageDB [save] rez={0}",rez);
            }
        }

        public Order FindOne(int id)
        {
            throw new System.NotImplementedException();
        }

        public IEnumerable<Order> FindAll()
        {
            throw new System.NotImplementedException();
        }

        public void Remove(int id)
        {
            throw new System.NotImplementedException();
        }

        public void Update(int id, Order e)
        {
            throw new System.NotImplementedException();
        }
    }
}